### DISCUSSION TITLE
How to Handle Long Filename?

### DISCUSSION THREAD

**Discussion by WB-Student** [abraunegg/onedrive#2552] [Q&A] [answered]
Hello all, 

A huge thank you to @abraunegg for this OneDrive client. I am relatively new to Linux and this tool has been the single biggest factor in allowing me to daily drive Linux almost exclusively. Additionally, the documentation and resources on the Git Hub for this project are incredibly helpful. 

I recently discovered, however, that there are a handful of files with long filenames that are not syncing. Specifically, these files exist without issue in OneDrive on my Mac and when accessing OneDrive through the browser. Yet, they are not syncing to my Linux machine and I receive notices that their filenames are too long.

Is there a way to make the OneDrive client accept these files with long filenames?

Any help is most appreciated. 

Thank you.

**Comment by abraunegg** [comment, marked as answer]
@WB-Student 
Nothing can be done as the API has hard limits of 400 characters (business and sharepoint) and 430 for personal - this is all documented as hard limits here:

https://support.microsoft.com/en-au/office/restrictions-and-limitations-in-onedrive-and-sharepoint-64883a5d-228e-48f5-b3d2-eb39e07630fa#filenamepathlengths

There is nothing that can be done here other than you changing your path / folder structure .. sorry

**Reply by WB-Student** [reply]
Hi @abraunegg,

Thank you for getting back to me so quickly. That is disappointing to hear about the API limits and, as you said, there is nothing we can do.

I suppose my renaming those files to something much shorter would fix their not syncing onto my Linux machine through the OneDrive Client for Linux. Thankfully, it is only four files.

Just curious: I am running the OneDrive Client for Linux in `--monitor` mode with an open terminal window. As I am also a OneNote user, this leads to _several_ of the 

WARNING: OneDrive API inconsistency - this file does not have any hash: 

notices you have addressed [elsewhere](https://github.com/abraunegg/onedrive/discussions/2441). 

While I am aware not to worry about these WARNING: API notices, the filename length errors, such as
```
ERROR: The local file system returned an error with the following message:
  Error Message:    Cannot open file 'WB/Postdoc/Charleston/VAQS/Literature/(Success in Academic Surgery) Justin B. Dimick, Caprice C. Greenberg (eds.) - Success in Academic Surgery_ Health Services Research-Springer-Verlag London (2014) (dragged).pdf.partial' in mode 'wb' (File name too long)
```
tend to get mixed in with the API warnings. Is there a way to specifically call for the ERRORs so I can more easily identify any files that are too long/need renamed?

Thank you.

**Reply by abraunegg** [reply]
@WB-Student 
Your best method is to run the client in `--verbose` mode + via a normal systemd service (which runs --monitor) + enable logging (please read: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#client-activity-log)

Once you have logging enabled, you should be able to use `grep` to search the logfile real-time:
```
tail -f /path/to/logfile | grep 'File name too long'
```

**Reply by WB-Student** [reply]
Hi @abraunegg,

Thank you for the suggestion. After reading over documentation for running OneDrive Client for Linux as a systemd service, will enabling that lead to the OneDrive Client for Linux starting and running automatically on system startup? While I prefer having OneDrive run and sync _most_ of the time I am using my machine, I do not want it to automatically run on startup (in case I messed something up or there was a disaster with OneDrive on another machine I could boot my Linux machine and keep those local files intact). I would prefer choosing when that system runs (or doesn't). 

I hope that makes sense.

**Reply by abraunegg** [reply]
@WB-Student 
> After reading over documentation for running OneDrive Client for Linux as a systemd service, will enabling that lead to the OneDrive Client for Linux starting and running automatically on system startup?

Yes it will.

> I do not want it to automatically run on startup

In that case disable and remove the systemd service operation, but keep logging enabled

**Reply by WB-Student** [reply]
Hi @abraunegg,

Got it.

Thank you so much for your help. I have a _much_ better understanding of this incredibly useful tool you built.

**Comment by MartinMiller23** [comment]
I use Long Path Tool for this problem. It makes it easy for the drive to accept the longname files.

**Comment by Rickeymarchram** [comment]
I use longpath tool for handling path too long files.

**Reply by abraunegg** [reply]
2nd comment saying the same thing - advertising a tool which costs money.

**Comment by John-davis78** [comment]
Try using Long Path Tool. It is effective.

**Reply by abraunegg** [reply]
3rd comment saying the same thing - advertising a tool which costs money.
